### PR TITLE
fix: Add support for jackson with java time modules

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Add support for java Instant new Java8 Date API datatype modules in Jackson. 

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/28350

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [ ] Manual
- [ ] JUnit

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 1]
```diff
@@ -337,7 +338,9 @@ private Mono<ApplicationTemplate> uploadCommunityTemplateToCS(CommunityTemplateU
         String authHeader = "Authorization";
         String payload;
         try {
-            ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+            ObjectMapper mapper = new ObjectMapper();
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:341]: Can we make use of `SerializationUtils` from the interfaces module to instantiate this? Let's try to make our serialization happen using one common set of rules all the time.

File: app/server/appsmith-server/pom.xml [hunk 2]
```diff
@@ -427,6 +427,14 @@
             <version>2.14.2.Final</version>
             <scope>test</scope>
         </dependency>
+
+        <!-- https://mvnrepository.com/artifact/com.fasterxml.jackson.datatype/jackson-datatype-jsr310 -->
```
Comment (nidhi-nair) [inline app/server/appsmith-server/pom.xml:431]: This is getting imported using the interfaces modules already, using serialization utils should manage this for us.

Comment (AnaghHegde) [reply app/server/appsmith-server/pom.xml:431]: Thanks for highlighting this. I was not aware of this. Let me make this change :)

**Review by nidhi-nair — DISMISSED** [review]

**Review by nidhi-nair — APPROVED** [review]

**Pull request merged**